*: use testutils.SetVModule

stevendanna opened the pull request:
This replaces most uses of log.SetVModule in tests with testutils.SetVModule, which handles resetting the vmodule back to its original value.

Epic: none
Release note: None

stevendanna committed:
*: use testutils.SetVModule

This replaces most uses of log.SetVModule in tests with
testutils.SetVModule, which handles resetting the vmodule back to its
original value.

Epic: none
Release note: None

tbg reviewed (approved):
I don't want to look a gift horse in the mouth - this is a great change, thank you - but I can't help wondering why `testutils.SetVModule` returns a closure when it could instead call `t.Cleanup(...)` and removes the accidental footgun that is

```
defer testutils.SetVModule(t, "asd") // oops
```

> @tbg reviewed 17 of 17 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154934#-:-OayMJPE6ZkresbHHFIE:b9nto75)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @andyyang890, @jeffswenson, @pav-kv, and @xinhaoz)



<!-- Sent from Reviewable.io -->

stevendanna commented:
I agree that is a foot gun. The main reason I didn't reach for t.Cleanup is that I don't think of that function often :D.  I think the main downside to me is that it does require some testing.T specific knowledge to know when that cleanup runs when mixed with other APIs that use defer.  

I'll think about it when looking at these test failures.

RaduBerinde commented:
+1 on using Cleanup. We don't think of it because we don't use it much, but we should use it more.

stevendanna committed:
testutils: use t.Cleanup in SetVModule

This avoids the potential for someone to forget to defer the returned
func.

Epic: none
Release note: None

stevendanna commented:
Doneburger.
